gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java

```diff
@@ -277,27 +277,46 @@ public void close() throws IOException {
   /**
    * Creates the given topic with the given name.
    *
+   * Sample code:
+   * <pre><code>
+   * try (PublisherApi publisherApi = PublisherApi.createWithDefaults()) {
+   *   String formattedName = PublisherApi.formatTopicName("[PROJECT]", "[TOPIC]");
+   *   Topic response = publisherApi.createTopic(formattedName);
+   * }
+   * </code></pre>
+   *
    * <!-- manual edit -->
    * <!-- end manual edit -->
    *
-   * @param name The name of the topic. It must have the format
+   * @param formattedName The name of the topic. It must have the format
    * `"projects/{project}/topics/{topic}"`. `{topic}` must start with a letter,
    * and contain only letters (`[A-Za-z]`), numbers (`[0-9]`), dashes (`-`),
    * underscores (`_`), periods (`.`), tildes (`~`), plus (`+`) or percent
    * signs (`%`). It must be between 3 and 255 characters in length, and it
    * must not start with `"goog"`.
    * @throws com.google.api.gax.grpc.ApiException if the remote call fails
    */
-  public final Topic createTopic(String name) {
-    Topic request = Topic.newBuilder().setName(name).build();
-
+  public final Topic createTopic(String formattedName) {
+    TOPIC_PATH_TEMPLATE.validate(formattedName);
```

Comment:
Splitting the formattedName into two params would be possible, but I think the output would be odd in some other cases. For example when accepting two formatted parameters like a subscription and a topic the user would need to pass in the same project in two different parameters.